### PR TITLE
Various changes, refactorings, and fixes 🔧

### PR DESCRIPTION
**Pull request by kevinlitchfield**
Main changes:

- Reformat values in `time` column (concatenating values in `date` column) so that they can be imported to Redshift using the `timestamp` data type. Redshift implements only a subset of Postgres data types, not including any `time` types.
- Improve error handling
- Use class methods for `SimpleLogger`
- Fix bug with incorrectly encoded characters. Very occasionally, `String#match` thought that URLs were incorrectly encoded when it was checking to see if `-ctx-` was present. Just in that one place (`LogFileCopier::LogReformatter::LogEntry#valid?`) call `String#scrub` to remove incorrectly encoded characters. 

Also:

- Refactor some stuff
- Reorganize files
- Fix typos
- Add Rubocop

### CODE REVIEW

File: test.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Bundler.require(:production, :test)
+Dotenv.load
+
+task = ARGV[0]
+log_file = ARGV[1]
+
+if task == 'rewrite'
+  LogFileCopier::LogReformatter.new(log_file).execute
+elsif task == 'import'
+  RedshiftImporter.new.import_new_log_files_into_redshift(test: true)
```
Comment (ksheurs) [inline test.rb]: what does `test: true` do here?

File: lib/log_file_copier/log_reformatter/log_entry.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  def initialize(raw_text)
+    @raw_text = raw_text
+  end
+
+  # Redshift doesn't have a time type, only a timestamp type. So we need to concatenate date with time for the second column.
+  def format_timestamp!
+    date = raw_text[/.+?\t/].strip
+
+    @raw_text.sub!(/(?<=\t).+?(?=\t)/) do |time|
+      date + ' ' + time
```
Comment (ksheurs) [inline lib/log_file_copier/log_reformatter/log_entry.rb]: Do we need to be concerned with timezones or anything?

**Comment by ksheurs** [comment]
Overall this looks great, Kevin. Let's ensure we are using our existing Redshift instance for cost savings + use existing Looker connections. Also worth noting that Vimeo's BI team is pulling data from us (and they are already hooked in to our Redshift instance).

File: test.rb (hunk 1 above)
Comment (kevinlitchfield) [inline test.rb]: Causes files on S3 not to be archived after import to Redshift. Was running some manual tests where that was useful. Added a couple comments in `test.rb` explaining.

File: lib/log_file_copier/log_reformatter/log_entry.rb (hunk 2 above)
Comment (kevinlitchfield) [inline lib/log_file_copier/log_reformatter/log_entry.rb]: No timezone is provided in the logs. According to [Akamai Log Delivery Service User Guide.pdf](https://github.com/vhx/import-akamai-logs-into-redshift/files/832293/Akamai.Log.Delivery.Service.User.Guide.pdf), the time in log file names is GMT, so I'm guessing the `time` field is also GMT. If we don't know for sure, would probably be worth a question to Fabian.

**Comment by kevinlitchfield** [comment]
@ksheurs Thanks for the feedback! The specific Redshift instance to be used is identified in `.env`, so I don't think anything has to change in the code for that. The only hardcoded assumption is that the table is named `akamai_logs`. Anyway, I've added a note to the logs explaining the importance of using the existing Redshift instance.

As you suggested, I've expanded the readme: https://github.com/vhx/import-akamai-logs-into-redshift/blob/003883d7faca9873f698918a71904d8c722224ec/README.md Let me know if it looks to you like there's anything missing.

File: lib/log_file_copier/log_reformatter/log_entry.rb (hunk 2 above)
Comment (ksheurs) [inline lib/log_file_copier/log_reformatter/log_entry.rb]: Yea docs seem to say "All time periods referenced in the Aggregation discussion are GMT." Probably fine!

Comment (kevinlitchfield) [inline lib/log_file_copier/log_reformatter/log_entry.rb]: OK, thanks.

**Pull request merged**